dpar/src/features/input_layers.rs

```diff
@@ -302,18 +302,19 @@ impl InputVectorizer {
                 let head = addr_head.get(state);
                 let dependent = addr_dependent.get(state);
                 if let (Some(head), Some(dependent)) = (head, dependent) {
-                    let association = self.assoc_strength(&head, &dependent, &deprel);
-                    non_lookup_slice[idx] = association;
+                    let (association, found) = self.assoc_strength(&head, &dependent, &deprel);
+                    non_lookup_slice[idx * 2] = association;
+                    non_lookup_slice[idx * 2 + 1] = found;
                 }
             }
         }
     }
 
-    fn assoc_strength(&self, head: &str, dependent: &str, deprel: &str) -> f32 {
+    fn assoc_strength(&self, head: &str, dependent: &str, deprel: &str) -> (f32, f32) {
```

Comment:
Tuples with elements of the same type can get confusing, I would return an `Option<f32>` (what you get out of `get`). Then when filling the array, you could set the value to `0f32` or `1f32` with something along the lines of:

```rust
non_lookup_slice[idx * 2] = association.unwrap_or(0f32);
non_lookup_slice[idx * 2 + 1] = association.map_or(0f32, |_| 1f32);
```